superset/common/query_context_processor.py

```diff
@@ -328,7 +378,25 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                     "when using a Time Comparison."
                 )
             )
-        for offset in time_offsets:
+
+        columns = df.columns
+        time_grain = self.get_time_grain(query_context, query_object) or TimeGrain.DAY
+        join_column_producer = config["TIME_GRAIN_JOIN_COLUMN_PRODUCERS"].get(
+            time_grain
+        )
+        use_aggregated_join_column = (
+            time_grain in AGGREGATED_JOIN_GRAINS or join_column_producer
+        )
```

Comment:
Ok, this is a total mininit and functionally they're equivalent, but for readability I'd place the check for `join_column_producer` first, as it takes precedent in downstream logic (this code is generally pretty heavy on the eyes, so any improved readability helps):
```suggestion
        use_aggregated_join_column = (
            join_column_producer  or time_grain in AGGREGATED_JOIN_GRAINS
        )
```